stdlib/public/core/StringUnicodeScalarView.swift

```diff
@@ -115,39 +115,27 @@ extension String {
     ///     print(hearts.unicodeScalars[j].value)
     ///     // Prints "9829"
     public struct Index : Comparable {
-      public init(_ _position: Int, _ _core: _StringCore) {
+      public init(_ _position: Int) {
```

Comment:
Please make `_position` a required label, and mark the API with `public // @testable`.  (Are we using it in tests?  If not, please just make it internal).

We can't use the name `init(_:)` because this will allow the users to construct the index from an `Int`.  Previously the initializer also had a `_StringCore` argument, which users could not create, so this initializer was effectively inaccessible.